mlrun/runtimes/base.py

```diff
@@ -165,6 +165,14 @@ def __init__(self, metadata=None, spec=None):
         self.verbose = False
         self._enriched_image = False
 
+    @staticmethod
```

Comment:
```suggestion
    @classmethod
```
as in subclasses, otherwise there is no `cls` parameter.